packages/react-form-hooks/src/hooks/field/field.ts

```diff
@@ -0,0 +1,196 @@
+ *
+ * ```typescript
+ * const title = useField('default title');
+ * return (<TextField label="Title" {...title} />);
+ * ```
+ *
+ * @param config - The default value of the input, or a configuration object specifying both the value and validation config.
+ * @param validationDependencies - An array of values for determining when to regenerate the field's validation callbacks. Any value that is referenced by a validator other than those passed into it should be included.
+ * @returns A `Field` object representing the state of your input. It also includes functions to manipulate that state. Generally, you will want to pass these callbacks down to the component or components representing your input.
+ *
+ * @remarks
+ * **Reinitialization:** If the `value` property of the field configuration changes between calls to `useField`,
+ * the field will be reset to use it as it's new default value.
+ *
+ * **Imperative methods:** The returned `Field` object contains a number of methods used to imperatively alter it's state.
+ * These should only be used as escape hatches where the existing hooks and components do not make your life easy,
+ * or to build new abstractions in the same vein as `useForm`, `useSubmit` and friends.
+ */
+export function useField<Value = string>(
+  input: FieldConfig<Value> | Value,
+  dependencies: unknown[] = [],
+): Field<Value> {
+  const {value, validates} = normalizeFieldConfig(input);
+  const validators = normalizeValidation(validates);
+
+  const [state, dispatch] = useFieldReducer(value);
+
+  const reset = useCallback(() => dispatch(resetAction()), [dispatch]);
+  const newDefaultValue = useCallback(
+    value => dispatch(newDefaultAction(value)),
+    [dispatch],
+  );
+
+  const runValidation = useCallback(
+    (value: Value = state.value) => {
+      const error = validators
+        .map(check => check(value, {}))
+        .filter(value => value != null);
+
+      if (error && error.length > 0) {
+        const [firstError] = error;
+        dispatch(updateErrorAction(firstError));
+        return firstError;
+      }
+
+      dispatch(updateErrorAction(undefined));
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [state.value, ...dependencies],
+  );
+
+  const onChange = useCallback(
+    (value: Value | ChangeEvent) => {
+      const normalizedValue = isChangeEvent(value) ? value.target.value : value;
+
+      dispatch(updateAction(normalizedValue) as any);
+
+      if (state.error) {
+        runValidation(normalizedValue as Value);
+      }
+    },
+    [runValidation, state.error, dispatch],
+  );
+
+  const setError = useCallback(value => dispatch(updateErrorAction(value)), [
+    dispatch,
+  ]);
+
+  const onBlur = useCallback(
+    () => {
+      if (state.touched === false && state.error == null) {
+        return;
+      }
+
+      runValidation();
+    },
+    [runValidation, state.touched, state.error],
```

Comment:
Do you actually need to reference both fields on `state`, or would it be enough to just invalidate when `state` changes?

Reply:
I think this is better because changing `state.value` would change the whole object but we don't actually need regenerate this in that case